### PR TITLE
fix: rtl support added to mega menu

### LINKED ISSUE #2039: Mega menu in RTL mode is wrongly rendered
#### Is this a bug, enhancement, or feature request?
bug

#### Briefly describe your proposal.
Screen Shot 2020-02-20 at 6 27 19 PM

### PR DESCRIPTION
**Pull request by rengare**
#### Please provide a link to the associated issue.
Fixes #2039 

#### Please provide a brief summary of this pull request.
RTL support added to keyboard events and visuals 

### Before:
image

### After:
image

#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done

### CODE REVIEW

**Comment by rengare** [comment]
arrow Icons are not visible in IE 11 
image

File: libs/core/src/lib/mega-menu/mega-menu.component.ts [hunk 1]
```diff
@@ -27,4 +32,12 @@ import { ChangeDetectionStrategy, Component, ViewEncapsulation } from '@angular/
     encapsulation: ViewEncapsulation.None,
     changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class MegaMenuComponent {}
+export class MegaMenuComponent {
+    dir$: BehaviorSubject<string> = new BehaviorSubject(rtl);
+
+    constructor(@Optional() private rtlService: RtlService) {
+        if (rtlService) {
+            rtlService.rtl.subscribe(isRtl => this.dir$.next(isRtl ? rtl : ltr));
```
Comment (mikerodonnell89) [inline libs/core/src/lib/mega-menu/mega-menu.component.ts]: Why use the ltr/rtl consts here when we can just use the boolean from the service?

File: libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts [hunk 2]
```diff
@@ -92,33 +94,21 @@ export class MegaMenuItemComponent implements AfterContentInit, OnDestroy, Defau
     constructor(
         private elRef: ElementRef,
         private menuKeyboardService: MenuKeyboardService,
-        private changeDetectionRef: ChangeDetectorRef
-    ) {}
+        private changeDetectionRef: ChangeDetectorRef,
+        @Optional() private rtlService: RtlService
+    ) {
+        if (rtlService) {
```
Comment (salarenko) [inline libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts]: - Unsubscribe
- Can we move this to separate method?

File: libs/core/src/lib/mega-menu/mega-menu.component.ts (hunk 1 above)
Comment (salarenko) [inline libs/core/src/lib/mega-menu/mega-menu.component.ts]: - Unsubscribe
- We can simplify this example and use Observable

File: libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts [hunk 3]
```diff
@@ -254,6 +244,59 @@ export class MegaMenuItemComponent implements AfterContentInit, OnDestroy, Defau
         this.subItems.forEach((item: MegaMenuSubitemDirective, index: number) => item.keyDown
             .pipe(takeUntil(this.onDestroy$))
             .subscribe((keyboardEvent: KeyboardEvent) => this.handleSubListKeyDown(keyboardEvent, index)))
-        ;
+            ;
+    }
+
+    private _keyboardLtr(event: KeyboardEvent) {
+        switch (event.key) {
```
Comment (salarenko) [inline libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts]: I know this is not in the scope of the ticket, but can we by the way fix keyboard navigation on IE11?
IE11 doesn't have `key` property and supports old `keyCode`.

File: libs/core/src/lib/mega-menu/mega-menu.component.ts (hunk 1 above)
Comment (rengare) [reply libs/core/src/lib/mega-menu/mega-menu.component.ts]: done

File: libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts (hunk 3 above)
Comment (rengare) [reply libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts]: I fixed it partially (only left and right) there is still problem with Up and Down

File: libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts (hunk 2 above)
Comment (rengare) [reply libs/core/src/lib/mega-menu/mega-menu-item/mega-menu-item.component.ts]: done

**Pull request merged**